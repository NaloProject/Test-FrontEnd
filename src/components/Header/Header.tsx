import { FC } from 'react'
import { Roboto, Open_Sans } from 'next/font/google'
import clsx from 'clsx'

import { ReactComponent as FlameIcon } from '@Svg/flame-icon.svg'
import { ReactComponent as PaintingIcon } from '@Svg/painting-icon.svg'

const roboto = Roboto({
  subsets: ['latin'],
  weight: ['100', '300', '400', '500', '700', '900'],
})
const openSans = Open_Sans({ subsets: ['latin'] })

export type HeaderProps = {
  iconName: string
  className?: string
  subTitle: string
  title: string
}

const Icons: Record<string, FC> = {
  painting: PaintingIcon,
  flame: FlameIcon,
}

const Header: FC<HeaderProps> = ({ className, iconName, subTitle, title }) => {
  const Icon: FC | undefined = Icons[iconName]

  return (
    <header className={clsx(`flex flex-col`, className)}>
      <h1
        className={`flex align-middle ${roboto.className} text-slate-700 dark:text-slate-200`}
      >
        <span className="font-bold mr-2 text-[24px] sm:mr-2 sm:text-[28px] md:mr-3 md:text-[32px]">
          {title}
        </span>

        <span className="h-[32px] w-[20px] sm:w-[25px] md:w-[30px]">
          {Icon && <Icon />}
        </span>
      </h1>
      <h2
        className={`text-slate-400 dark:text-slate-600 text-[11px] sm:text-[12px] md:text-[14px] ${openSans.className}`}
      >
        {subTitle}
      </h2>
    </header>
  )
}

export default Header
