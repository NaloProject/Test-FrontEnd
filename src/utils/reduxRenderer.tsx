import { configureStore, EnhancedStore } from '@reduxjs/toolkit'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import { ReactElement, FC, PropsWithChildren } from 'react'
import { Provider } from 'react-redux'

import { combined, RootState } from '@Redux/store'
import INITIAL_STATE from './initialState'

type RenderWithReduxOptions = Omit<RenderOptions, 'queries'> & {
  initialState?: RootState
  store?: EnhancedStore
}

export const renderWithRedux = (
  ui: ReactElement,
  {
    initialState = INITIAL_STATE,
    store = configureStore({
      middleware: (getDefaultMiddleware) =>
        getDefaultMiddleware({
          immutableCheck: false,
          serializableCheck: false,
        }),
      preloadedState: initialState,
      reducer: combined,
    }),
    ...renderOptions
  }: RenderWithReduxOptions = {}
): RenderResult => {
  const Wrapper: FC<PropsWithChildren> = ({ children }) => (
    <Provider store={store}>{children}</Provider>
  )

  return render(ui, { wrapper: Wrapper, ...renderOptions })
}
