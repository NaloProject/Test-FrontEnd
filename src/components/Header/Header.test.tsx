import { renderWithRedux } from '@Utils/reduxRenderer'
import Header from './Header'

describe('Header', () => {
  it('should mount first', async () => {
    const { container } = renderWithRedux(
      <Header title="title" iconName="flame" subTitle="sub title" />
    )

    expect(container).toMatchSnapshot()
  })

  it('should render even if the icon is not found', async () => {
    const { container } = renderWithRedux(
      <Header title="title" iconName="wrongName" subTitle="sub title" />
    )

    expect(container).toMatchSnapshot()
  })
})
