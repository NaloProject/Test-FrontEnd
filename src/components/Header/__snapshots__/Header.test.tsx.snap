// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should mount first 1`] = `
<div>
  <header
    class="flex flex-col"
  >
    <h1
      class="flex align-middle className text-slate-700 dark:text-slate-200"
    >
      <span
        class="font-bold mr-2 text-[24px] sm:mr-2 sm:text-[28px] md:mr-3 md:text-[32px]"
      >
        title
      </span>
      <span
        class="h-[32px] w-[20px] sm:w-[25px] md:w-[30px]"
      >
        <svg />
      </span>
    </h1>
    <h2
      class="text-slate-400 dark:text-slate-600 text-[11px] sm:text-[12px] md:text-[14px] className"
    >
      sub title
    </h2>
  </header>
</div>
`;

exports[`Header should render even if the icon is not found 1`] = `
<div>
  <header
    class="flex flex-col"
  >
    <h1
      class="flex align-middle className text-slate-700 dark:text-slate-200"
    >
      <span
        class="font-bold mr-2 text-[24px] sm:mr-2 sm:text-[28px] md:mr-3 md:text-[32px]"
      >
        title
      </span>
      <span
        class="h-[32px] w-[20px] sm:w-[25px] md:w-[30px]"
      />
    </h1>
    <h2
      class="text-slate-400 dark:text-slate-600 text-[11px] sm:text-[12px] md:text-[14px] className"
    >
      sub title
    </h2>
  </header>
</div>
`;
